Elcano_C2_LowLevel/Elcano_C2_LowLevel.ino

```diff
@@ -236,119 +263,156 @@ void setup()
-        attachInterrupt(digitalPinToInterrupt(IRPT_TURN),  ISR_TURN_rise,  RISING);
-        //attachInterrupt(digitalPinToInterrupt(IRPT_RDR),   ISR_RDR_rise,   RISING);
-        attachInterrupt(digitalPinToInterrupt(IRPT_GO),    ISR_GO_rise,    RISING);
-        attachInterrupt(digitalPinToInterrupt(IRPT_ESTOP), ISR_ESTOP_rise, RISING);
-        //attachInterrupt(digitalPinToInterrupt(IRPT_RVS),   ISR_RVS_rise,   RISING);
-        attachInterrupt(digitalPinToInterrupt(IRPT_SWITCH), ISR_SWITCH_rise, RISING);
-        attachInterrupt(digitalPinToInterrupt(IRPT_MOTOR_FEEDBACK), ISR_MOTOR_FEEDBACK_rise, RISING);
-        //Print7headers(false);
-  //PrintHeaders();
+
+  // @ToDo: Remove any test and calibration code.
+  // //setupWheelRev(); // WheelRev4 addition
+  //CalibrateTurnAngle(32, 20);
+  //calibrationTime_ms = millis();
+
+  attachInterrupt(digitalPinToInterrupt(IRPT_TURN),  ISR_TURN_rise,  RISING);
+  //attachInterrupt(digitalPinToInterrupt(IRPT_RDR),   ISR_RDR_rise,   RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_GO),    ISR_GO_rise,    RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_ESTOP), ISR_ESTOP_rise, RISING);
+  //attachInterrupt(digitalPinToInterrupt(IRPT_RVS),   ISR_RVS_rise,   RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_SWITCH), ISR_SWITCH_rise, RISING);
+  attachInterrupt(digitalPinToInterrupt(IRPT_MOTOR_FEEDBACK), ISR_MOTOR_FEEDBACK_rise, RISING);
 }
-/*---------------------------------------------------------------------------------------*/
-void loop() {
-  //Serial.println("Start of loop");
-  SerialData Results;
 
-  // Save start time for performance report.
-  unsigned long startTime = micros();
-  // Get the next loop start time.
-  unsigned long nextTime = startTime + LOOP_TIME_US;
+/*---------------------------------------------------------------------------------------*/
 
-  digitalWrite(27, LOW);
-  digitalWrite(33, LOW);
-  startCapturingRCState();
+// Time at which this loop pass should end in order to maintain a
+// loop period of LOOP_TIME_US.
+unsigned long nextTime = micros();
+// Time at which we reach the end of loop(), which should be before
+// nextTime if we have set the loop period long enough.
+unsigned long endTime;
+// How much time we need to wait to finish out this loop pass.
+unsigned long delayTime;
 
-  unsigned long local_results[7];
-  //  PrintDone();
+// Inter-module communications data.
+SerialData Results;
 
-  while (micros() < nextTime &&
-         ~((RC_Done[RC_ESTP] == 1) && (RC_Done[RC_GO] == 1) && (RC_Done[RC_TURN] == 1) && (RC_Done[RC_RDR] == 1)))
-  {
-    ;  //wait
-  }
-  // got data;
-  for (int i = 0; i < 8; i++)
+void loop() {
+  // Get the next loop start time. Note this (and the micros() counter) will
+  // roll over back to zero after they exceed the 32-bit size of unsigned long,
+  // which happens after about 70 minues of operation. If the new nextTime
+  // value is <= LOOP_TIME_US, we've rolled over.
+  nextTime = nextTime + LOOP_TIME_US;
```

Comment:
If you look at the previous version you will see that this is **not a change** from what was present before.  This revision is for getting the boilerplate code into stable and reliable form, and is not intended to introduce functional changes.  Let's hold off on functional changes til after the cleanup.  Changing the time granularity does not affect the structure of the code, so altering it will be simple.

> range of 200 to 25 ms

If you mean that the loop interval should be allowed to *vary* over that range, then I disagree.  There are, or will be, things that depend on getting data, or having a chance to update actuator controls, at a constant pace.  But this is a discussion that can be had outside this PR, as it's not relevant to the bug fixes and cleanup here.